### DISCUSSION TITLE
Code listing for executable code chunks!

### DISCUSSION THREAD

**Discussion by aelmokadem** [quarto-dev/quarto-cli#6216] [Q&A] [answered]
Hi,

The quarto documentation mentions that we can reference code listings as in here https://quarto.org/docs/authoring/cross-references.html#code-listings. However, that seems to only work for static code chunks and not for executable code chunks! Is that correct and will that be a feature that will come to quarto soon? 

In the time being, is the only work around this is to have two copies of the code chunk, one executable and hidden and another static one with the listing reference?

Thank you.

Ahmed.

**Comment by cscheid** [comment, marked as answer]
> In the time being, is the only work around this is to have two copies of the code chunk, one executable and hidden and another static one with the listing reference?

Yes, that's the only currently supported workaround.

**Comment by mcanouil** [comment]
FYI the same question already answered with a working "knitr" example:

- #3626

**Reply by aelmokadem** [reply]
That's really helpful @mcanouil  thanks for sharing.
